mslib/msui/flighttrack.py

```diff
@@ -94,6 +94,34 @@ def seconds_to_string(seconds):
 TABLE_SHORT = [TABLE_FULL[_i] for _i in range(7)] + [TABLE_FULL[-1]] + [("", lambda _: "", False)] * 8
 
 
+def load_from_xml_data(xml_content, name="Flight track"):
+    try:
+        doc = xml.dom.minidom.parseString(xml_content)
+    except xml.parsers.expat.ExpatError as ex:
+        raise SyntaxError(str(ex))
+
+    ft_el = doc.getElementsByTagName("FlightTrack")[0]
+
+    waypoints_list = []
+    for wp_el in ft_el.getElementsByTagName("Waypoint"):
+
+        location = wp_el.getAttribute("location")
+        lat = float(wp_el.getAttribute("lat"))
+        lon = float(wp_el.getAttribute("lon"))
+        flightlevel = float(wp_el.getAttribute("flightlevel"))
+        comments = wp_el.getElementsByTagName("Comments")[0]
+        # If num of comments is 0(null comment), then return ''
+        if len(comments.childNodes):
+            comments = comments.childNodes[0].data.strip()
+        else:
+            comments = str('')
+
+        waypoints_list.append(Waypoint(lat, lon, flightlevel,
+                                       location=location,
+                                       comments=comments))
+    return waypoints_list
+
+
 class Waypoint(object):
```

Comment:
class Waypoint:

since python version 3 this is a replacement for (object) 
(this can also be done in a further refactoring)